src/libnotify/notification.cr

```diff
@@ -0,0 +1,74 @@
+require "./c"
+
+class Libnotify::Notification
+  property summary : String?
+  property body : String?
+  property icon_path : String?
+  property app_name : String
+  property timeout : Int32
+  property category : String?
+  property append : Bool
+  property transient : Bool
+  property urgency : Libnotify::C::Notifyurgency
+
+  def initialize(@summary = nil, @body = nil, @icon_path = nil,
+                 @timeout = -1, @category = nil, @urgency = Libnotify::C::Notifyurgency::NotifyUrgencyNormal,
+                 @append = false, @transient = true,
+                 @app_name = "default")
+    init_app!
+    @notify = C.notification_new @summary.to_s, @body.to_s, @icon_path.to_s
+    update
+  end
+
+  def initialize(@summary = nil, @body = nil, @icon_path = nil,
+                 @timeout = -1, @category = nil, @urgency = Libnotify::C::Notifyurgency::NotifyUrgencyNormal,
+                 @append = false, @transient = true,
+                 @app_name = "default", &block)
+    init_app!
+    @notify = C.notification_new @summary.to_s, @body.to_s, @icon_path.to_s
+    yield self
+    update
+  end
+
+  def show(gerror = nil)
+    C.notification_show @notify, gerror
```

Comment:
I wonder if we could call here `update` every time so need of chaining these calls `libnotify.update.show` will disappear.

Another solution would be to let `update` accept the same arguments (and block) as `Libnotify.new` like:

```crystal
notify = Libnotify.new
notify.update(body: "summary") do |n|
  n.timeout = 23
end
# no need to call `.show`
```

Which one do you like? :)